locale/pt-br/docs/guides/dont-block-the-event-loop.md

```diff
@@ -0,0 +1,488 @@
+---
+title: Não bloqueie o Event Loop (ou a Worker Pool)
+layout: docs.hbs
+---
+
+# Não bloqueie o Event Loop (ou a Worker Pool)
+
+## Você deve ler esse guia?
+Se você está escrevendo algo mais complicado que um breve script de linha de comando, ler este guia ajudará você a escrever aplicativos de maior desempenho e mais seguros.
+
+Este documento foi escrito com servidores Node em mente, mas os conceitos são aplicados para aplicações Node complexas também.
+Onde detalhes específicos do sistema operacional variam, este documento é centrado no Linux.
+
+## Resumo
+O Node.js executa código JavaScript no Event Loop (inicialização e callbacks), e oferece uma Worker Pool para manipular tarefas custusas como I/O de arquivo.
+Node escala bem, as vezes mais do que abordagens pesadas como Apache.
+O segredo da escalabilidade do Node é que ele usa um pequeno número de threads para manipular muitos clientes.
+Se o Node pode trabalhar com menos threads, ele poderá gastar mais tempo do seu sistema e memória trabalhando nos clientes em vez de disperdiçar recursos de espaço e tempo para as threads (memória e mudança de contexto).
+Mas pelo fato do Node ter poucas threads, você precisa estruturar sua aplicação para usá-las com sabedoria.  
+
+Aqui está um princípio básico para manter o servidor Node rápido: *Node é rápido quando o trabalho associado a cada cliente em um determinado momento é "pequeno"*.
+
+Isso se aplica a callbacks no Event Loop e tarefas na Worker Pool.
+
+## Por que eu devo evitar bloquear o Event Loop e a Worker Pool?
+O Node usa um pequeno número de threads para manipular muitos clientes.
+No Node existem dois tipos de threads: um Event Loop (também conhecido como main loop, main thread, event thread, etc.), e uma pool de `k` Workers em uma Worker Pool (também conhecido como threadpool)
+
+Se uma thread está levando muito tempo para excutar um callback (Event Loop) ou uma tarefa (Worker), nós a chamamos de "bloqueada".
+Enquanto uma thread está bloqueada trabalhando para um cliente, ela não pode lidar com requisições de outros clientes.  
+Isso fornece duas motivações para não bloquear o Event Loop nem a Worker Pool:
+
+1. Performance: Se você executar regularmente atividades pesadas em qualquer tipo de thread, o *throughput* (requisições por segundo) do seu servidor sofrerá.
+2. Segurança: Se for possível que para determinadas entradas uma de suas threads possam bloquear, um cliente malicioso pode enviar esse "evil input", para fazer suas threads bloquearem, e mantê-las trabalhando para outros clientes. Isso seria um ataque de [Negação de Serviço](https://en.wikipedia.org/wiki/Denial-of-service_attack)
```

Comment:
```suggestion
2. Segurança: Se for possível que para determinadas entradas uma de suas threads seja bloqueada, um cliente malicioso pode enviar esse "evil input", para fazer suas threads bloquearem, e mantê-las trabalhando para outros clientes. Isso seria um ataque de [Negação de Serviço](https://en.wikipedia.org/wiki/Denial-of-service_attack)
```

Reply:
I tried to improve this translation, what do you think?

Reply:
Where is it? I could not find it here

Reply:
Something like that **"Se for possível que, para determinadas entradas, uma de suas threads possa bloquear..."**

Reply:
I think this is weird because we already have the word "bloquear" like this:

> Segurança: Se for possível que para determinadas entradas uma de suas threads possam **bloquear**, um cliente malicioso pode enviar esse "evil input", para fazer suas threads **bloquearem**

So I guess it is repetitive...

Reply:
that's not weird, it was just my suggestion.

Reply:
I accepted your change, could you review please @khaosdoctor 

https://github.com/nodejs/nodejs.org/pull/2685/commits/06c7c9852044d1b9a937226aa633c24fafc1db6c

Reply:
Yeah! Sorry about the way I said it! I did not mean to be rude... I think my suggestion is a bit weird to... But I really don't know how to translate it in a better way...

Reply:
no problem bro :smile: